src/allocator/memserver_allocator.cpp

```diff
@@ -721,31 +1035,56 @@ int Memserver_Allocator::open_heap(uint64_t regionId) {
     if (heap == NULL) {
 
         // Heap is not open, open it now
-        int ret = memoryManager->FindHeap((PoolId)regionId, &heap);
-
+        int ret;
+        {
+            uint64_t start = NVMM_get_time();
+            ret = memoryManager->FindHeap((PoolId)regionId, &heap);
+            uint64_t end = NVMM_get_time();
+            uint64_t total = NVMM_time_diff_nanoseconds(start, end);
+            NVMM_PROFILE_ADD_TO_TOTAL_OPS(FindHeap, total);
+        }
         if (ret != NO_ERROR) {
             message << "heap not found";
             delete heap;
             throw Memserver_Exception(HEAP_NOT_OPENED, message.str().c_str());
         }
-        heap->Open();
-
+        {
+            uint64_t start = NVMM_get_time();
+            heap->Open();
+            uint64_t end = NVMM_get_time();
+            uint64_t total = NVMM_time_diff_nanoseconds(start, end);
+            NVMM_PROFILE_ADD_TO_TOTAL_OPS(Heap_Open, total);
+        }
+        uint64_t start1 = NVMM_get_time();
         pthread_mutex_lock(&heapMapLock);
+        uint64_t start2 = NVMM_get_time();
 
         // Heap opened now, Add this into map for future references.
         auto heapObj = heapMap->find(regionId);
         if (heapObj == heapMap->end()) {
-            heapMap->insert({regionId, heap});
+            heapMap->insert({ regionId, heap });
         } else {
             message << "Can not insert heap. regionId already found in map";
             pthread_mutex_unlock(&heapMapLock);
-            heap->Close();
+            {
+                uint64_t start = NVMM_get_time();
```

Comment:
Should we capture the time if profiling is not enabled ? Can we add these into macros ?